vulcan/isolation.py

```diff
@@ -90,12 +92,24 @@ def install(self, deps_dir: Union[str, bytes, 'PathLike[str]', 'PathLike[bytes]'
             '--use-pep517',
             '--target',
             str(deps_dir)] + requirements
-        subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        proc = await asyncio.create_subprocess_exec(*cmd, stderr=asyncio.subprocess.PIPE,
+                                                    stdout=asyncio.subprocess.DEVNULL)
+        _, err = await proc.communicate()
+        assert proc.returncode is not None
+        if proc.returncode != 0:
+            raise subprocess.CalledProcessError(returncode=proc.returncode, cmd=cmd, stderr=err)
 
-    def freeze(self, deps_dir: Union[str, bytes, 'PathLike[str]', 'PathLike[bytes]']
-               ) -> Dict[str, Requirement]:
+    async def freeze(self, deps_dir: Union[str, bytes, 'PathLike[str]', 'PathLike[bytes]']
+                     ) -> Dict[str, Requirement]:
         # list with the requirements.txt format only libraries installed in specifically this venv
         cmd = [self.context.env_exe, '-Im', 'pip', 'list', '--format=freeze', '--path', str(deps_dir)]
-        frozen = subprocess.check_output(cmd, encoding='utf-8')
-        reqs = [Requirement.parse(line) for line in frozen.split('\n') if line]
+
+        frozen = await asyncio.create_subprocess_exec(*cmd, stderr=asyncio.subprocess.DEVNULL,
+                                                      stdout=asyncio.subprocess.PIPE)
+
+        out, err = await frozen.communicate()
```

Comment:
don't you ignore stderr by setting `stderr=asyncio.subprocess.DEVNUL` ?

Reply:
yes, good call.